### PR TITLE
ENG-4041 Supported docker:// in generate-cr and API claims

### PR DESCRIPTION
**Pull request by zonia3000**

### CODE REVIEW

**Comment by zonia3000** [comment]
@cecchisandrone 

In the penultimate commit I moved the definition of the `TARBALL_PREFIX` constant from `custom-resource-service.ts` to `bundle-descriptor-constraints.ts` because it was circularly imported causing `yaml-bundle-descriptor-constraints.ts` to be loaded before `bundle-descriptor-constraints.ts`. This resulted in the `NAV_CONSTRAINTS` object to be undefined in `YAML_BUNDLE_DESCRIPTOR_CONSTRAINTS` and produced the following error 🤯:

image
I added a test case that would have failed in that situation.

In the last commit I also added a stricter validation for the nav `target` field.

File: README.md [hunk 1]
```diff
@@ -96,15 +96,15 @@ ARGUMENTS
   CLAIMNAME  Name of the API claim
 
 FLAGS
-  --bundle=<value>       Target Bundle Docker repository with the format <organization>/<repository> or
-                         <registry>/<organization>/<repository>
+  --bundle=<value>       Target Bundle Docker repository with the format (docker://)<organization>/<repository> or
```
Comment (cecchisandrone) [inline README.md]: I think optional values should be indicated with []

File: README.md [hunk 2]
```diff
@@ -166,8 +166,8 @@ USAGE
 
 FLAGS
   -d, --digest          Include Docker images digests
-  -i, --image=<value>   Name of the bundle Docker image with the format <organization>/<repository> or
-                        <registry>/<organization>/<repository>
+  -i, --image=<value>   Name of the bundle Docker image with the format (docker://)<organization>/<repository> or
+                        (docker://)<registry>/<organization>/<repository>
```
Comment (cecchisandrone) [inline README.md]: same

File: src/commands/api/add-ext.ts [hunk 3]
```diff
@@ -16,7 +17,7 @@ export default class AddExt extends Command {
     'Add an external API claim to the specified MFE component'
 
   static examples = [
-    '<%= config.bin %> <%= command.id %> mfe1 ms1-api --bundle registry.hub.docker.com/my-org/my-bundle --serviceName ms1'
+    '<%= config.bin %> <%= command.id %> mfe1 ms1-api --bundle docker://registry.hub.docker.com/my-org/my-bundle --serviceName ms1'
```
Comment (cecchisandrone) [inline src/commands/api/add-ext.ts]: let's keep simplified examples

File: src/models/bundle-descriptor-constraints.ts [hunk 4]
```diff
@@ -39,11 +39,12 @@ export const INVALID_NAME_MESSAGE =
   'Name components may contain lowercase letters, digits and separators. A separator is defined as a period, one or two underscores, or one or more dashes. A name component may not start or end with a separator.'
 export const INVALID_VERSION_MESSAGE =
   'Version may contain lowercase and uppercase letters, digits, underscores, periods and dashes. Version may not start with a period or a dash.'
-export const ALLOWED_BUNDLE_WITHOUT_REGISTRY_REGEXP = /^[\w-]+\/[\w-]+$/
+export const TARBALL_PREFIX = 'docker://'
```
Comment (cecchisandrone) [inline src/models/bundle-descriptor-constraints.ts]: why TARBALL? better DOCKER_PREFIX

**Review by cecchisandrone — APPROVED** [review]

**Pull request merged**